clang/lib/StaticAnalyzer/Core/CallEvent.cpp

```diff
@@ -748,6 +747,22 @@ SVal CXXInstanceCall::getCXXThisVal() const {
   return ThisVal;
 }
 
+const CXXRecordDecl *CXXInstanceCall::getDeclForDynamicType() const {
+  const MemRegion *R = getCXXThisVal().getAsRegion();
+  if (!R)
+    return nullptr;
+
+  DynamicTypeInfo DynType = getDynamicTypeInfo(getState(), R);
+  if (!DynType.isValid())
+    return nullptr;
+
+  QualType Ty = DynType.getType()->getPointeeType();
+  if (Ty.isNull())
```

Comment:
Previously we had an assertion here. Is this invariant of "DynamicTypeInfo" is not always a pointer not actually true?

Reply:
My reasoning was that while it was asserted here, it may not apply in general. Given that I abstract this code in a separate function, I may shouldn't enforce this invariant.
Although, the `assert˙wouldn't fire in any of the tests we have.
What I could do, is to have an assert and also a graceful return in that case. That way we would get the benefit of both world.

Reply:
Sounds good! The assert message made me think that this was a global invariant, not specific to the call site.